tests/e2e/query_modules_storage_modes/test_query_modules_storage_modes.py

```diff
@@ -0,0 +1,281 @@
+# Copyright 2023 Memgraph Ltd.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+# License, and you may not use this file except in compliance with the Business Source License.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0, included in the file
+# licenses/APL.txt.
+
+# isort: off
+from multiprocessing import Process
+import sys
+import pytest
+
+from common import cursor, connect
+
+import time
+
+
+def modify_graph(query):
+    subprocess_cursor = connect()
+
+    time.sleep(0.5)
+
+    subprocess_cursor.execute(query)
+
+
+@pytest.mark.parametrize("api", ["c", "cpp", "python"])
+def test_function_delete_result(cursor, api):
+    cursor.execute("STORAGE MODE IN_MEMORY_ANALYTICAL;")
+
+    deleter = Process(
+        target=modify_graph,
+        args=("MATCH (m:Component {id: 'A7422'})-[e:PART_OF]->(n:Component {id: '7X8X0'}) DELETE e;",),
+    )
+    deleter.start()
+
+    cursor.execute(f"MATCH (m)-[e]->(n) RETURN {api}_api.pass_relationship(e);")
+
+    deleter.join()
+
+    result = cursor.fetchall()
+
+    assert len(result) == 1 and len(result[0]) == 1 and result[0][0].type == "DEPENDS_ON"
+
+
+@pytest.mark.parametrize("api", ["c", "cpp", "python"])
+def test_function_delete_only_result(cursor, api):
+    cursor.execute("STORAGE MODE IN_MEMORY_ANALYTICAL;")
+
+    cursor.execute("MATCH (m:Component {id: '7X8X0'})-[e:DEPENDS_ON]->(n:Component {id: 'A7422'}) DELETE e;")
+
+    deleter = Process(
+        target=modify_graph,
+        args=("MATCH (m:Component {id: 'A7422'})-[e:PART_OF]->(n:Component {id: '7X8X0'}) DELETE e;",),
+    )
+    deleter.start()
+
+    cursor.execute(f"MATCH (m)-[e]->(n) RETURN {api}_api.pass_relationship(e);")
+
+    deleter.join()
+
+    result = cursor.fetchall()
+
+    assert len(result) == 1 and len(result[0]) == 1 and result[0][0] is None
+
+
+@pytest.mark.parametrize("api", ["c", "cpp", "python"])
+def test_procedure_delete_result(cursor, api):
+    cursor.execute("STORAGE MODE IN_MEMORY_ANALYTICAL;")
+
+    deleter = Process(
+        target=modify_graph,
+        args=("MATCH (n {id: 'A7422'}) DETACH DELETE n;",),
+    )
+    deleter.start()
+
+    cursor.execute(
+        f"""MATCH (n)
+        CALL {api}_api.pass_node_with_id(n)
+        YIELD node, id
+        RETURN node, id;"""
+    )
+
+    deleter.join()
+
+    result = cursor.fetchall()
+
+    assert len(result) == 1 and len(result[0]) == 2 and result[0][0].properties["id"] == "7X8X0"
```

Comment:
Why do we have `len(result[0]) == 2` and check just the first one - `result[0][0]`. What about `result[0][1]`?

Reply:
The second field (`result[0][1]`) is the internal ID of the node that’s passed. I don’t evaluate its value because internal node IDs can vary (e.g. if we add more tests that execute before this one, the ID will be higher).